### PR TITLE
Add a warning incase GCC is used.

### PR DESCRIPTION
**Pull request by Tylertron1998**
Ubuntu seems to default to GCC, so we should warn the user in case this happens.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -64,7 +64,7 @@ Then, clone and build NovelRT:
 git clone https://github.com/NovelRT/NovelRT.git
 mkdir -p NovelRT/build
 cd NovelRT/build
-cmake ..
```
Comment (RubyNova) [inline README.md:67]: This is not what I meant.

This is also technically wrong.

Just note we don't support GCC at this time in plain english somewhere appropriate?

**Review by RubyNova — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Tylertron1998) [reply README.md:67]: Can you clarify what you mean by this? It ran perfectly fine on my machine - also yeah, adding some explanation is probably a good idea.

Comment (RubyNova) [reply README.md:67]: the official CXX compiler flag value is `clang++` and this is somewhat a hack on Linux. You'd usually change those env vars fully if you want to use clang.

File: README.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ sudo apt install clang cmake doxygen graphviz g++ libfreetype-dev libglfw3-dev \
     libspdlog-dev python3-glad
 ```
 Then, clone and build NovelRT:
+> **WARNING:** we currently do not support GCC as a compiler option. Expect errors/bugs if you try to use it - consider clang instead.
```
Comment (RubyNova) [inline README.md:63]: maybe instead of "consider" we could say "supported compilers are MSVC and clang for their respective platforms."?

**Comment by Tylertron1998** [comment]
after some internal discussion on the discord server - we have decided instead to just note what the supported route should be. While giving exact instructions would be nice, that can be difficult given the vast variety of distros/operating systems/setups.

**Review by RubyNova — APPROVED** [review]
LGTM
